src/liftdrag_plugin/advanced_liftdrag_plugin.cpp

```diff
@@ -0,0 +1,641 @@
+  this->upward.Normalize();
+
+  //Get reference area
+  if (_sdf->HasElement("area"))
+    this->area = _sdf->Get<double>("area");
+
+  //Get atmospheric density
+  if (_sdf->HasElement("air_density"))
+    this->rho = _sdf->Get<double>("air_density");
+
+  // Check if link exists
+  if (_sdf->HasElement("link_name"))
+  {
+    sdf::ElementPtr elem = _sdf->GetElement("link_name");
+    // GZ_ASSERT(elem, "Element link_name doesn't exist!");
+    std::string linkName = elem->Get<std::string>();
+    this->link = this->model->GetLink(linkName);
+    // GZ_ASSERT(this->link, "Link was NULL");
+
+    if (!this->link)
+    {
+      gzerr << "Link with name[" << linkName << "] not found. "
+        << "The AdvancedLiftDragPlugin will not generate forces\n";
+    }
+    else
+    {
+      this->updateConnection = event::Events::ConnectWorldUpdateBegin(
+          boost::bind(&AdvancedLiftDragPlugin::OnUpdate, this));
+    }
+  }
+
+  if (_sdf->HasElement("robotNamespace"))
+  {
+    namespace_ = _sdf->GetElement("robotNamespace")->Get<std::string>();
+  } else {
+    gzerr << "[gazebo_advanced_liftdrag_plugin] Please specify a robotNamespace.\n";
+  }
+  node_handle_ = transport::NodePtr(new transport::Node());
+  node_handle_->Init(namespace_);
+
+  if (_sdf->HasElement("topic_name")) {
+      const auto lift_force_topic = this->sdf->Get<std::string>("topic_name");
+      lift_force_pub_ = node_handle_->Advertise<physics_msgs::msgs::Force>("~/" + lift_force_topic);
+      gzdbg << "Publishing to ~/" << lift_force_topic << std::endl;
+  }
+
+  if (_sdf->HasElement("windSubTopic")){
+    this->wind_sub_topic_ = _sdf->Get<std::string>("windSubTopic");
+    wind_sub_ = node_handle_->Subscribe("~/" + wind_sub_topic_, &AdvancedLiftDragPlugin::WindVelocityCallback, this);
+  }
+
+}
+
+/////////////////////////////////////////////////
+void AdvancedLiftDragPlugin::OnUpdate()
+{
+  GZ_ASSERT(this->link, "Link was NULL");
+  // get linear velocity at ref_pt in inertial frame
+#if GAZEBO_MAJOR_VERSION >= 9
+  ignition::math::Vector3d vel = this->link->WorldLinearVel(this->ref_pt) - wind_vel_;
+  const common::Time current_time = this->world->SimTime();
+#else
+  ignition::math::Vector3d vel = ignitionFromGazeboMath(this->link->GetWorldLinearVel(this->ref_pt)) - wind_vel_;
+  const common::Time current_time = this->world->GetSimTime();
+#endif
+  const double dt = (current_time - this->last_pub_time).Double();
+
+  if (vel.Length() <= 0.01)
+    return;
+
+  // pose of body
+#if GAZEBO_MAJOR_VERSION >= 9
+  ignition::math::Pose3d pose = this->link->WorldPose();
+#else
+  ignition::math::Pose3d pose = ignitionFromGazeboMath(this->link->GetWorldPose());
+#endif
+
+  //Define body frame: X forward, Z downward, Y out the right wing
+  ignition::math::Vector3d bodyX = pose.Rot().RotateVector(this->forward);
+  ignition::math::Vector3d bodyZ = -1*(pose.Rot().RotateVector(this->upward));
+  ignition::math::Vector3d bodyY =  bodyZ.Cross(bodyX);
```

Comment:
```suggestion
  ignition::math::Vector3d body_x_axis = pose.Rot().RotateVector(this->forward);
  ignition::math::Vector3d body_z_axis = -1*(pose.Rot().RotateVector(this->upward));
  ignition::math::Vector3d body_y_axis =  body_z_axis.Cross(body_x_axis);
```

just a bit more clear what these vectors are